R/readTechCosts.R

```diff
@@ -59,29 +79,470 @@ readTechCosts <- function(subtype = "PowerAndHeat") { # nolint
+
+      dfp <- dfp[!is.na(dfp$value), ]
+
+      x <- as.quitte(dfp)
+
   } else if (subtype == "DomesticEnergy") {
-      x <- read.csv("domestic_energyf2.csv")
-      names(x) <- c("tech", "appliances", "sector", "value", "category", "measurement", "variant", "unit")
-      x[["value"]] <- as.numeric(x[["value"]])
-      x <- as.quitte(x)
+      df <- read_excel("REF2020_Technology Assumptions_Energy.xlsx", sheet = "Domestic", range = "A5:H69")
+
+      df = df[-c(1,19:22,42:47,54:57),]
+
+      index_of_NA <- which(is.na(df[,3]))
+
+      df$category_of_technology <- NA
+      df <- as.data.frame(df)
+
+      df[seq(from=index_of_NA[1], to=(index_of_NA[2]-1)) , 9] <- df[index_of_NA[1],1]
+      df[seq(from=index_of_NA[2], to=(index_of_NA[3]-1)) , 9] <- df[index_of_NA[2],1]
+      df[seq(from=index_of_NA[3], to=(index_of_NA[4]-1)) , 9] <- df[index_of_NA[3],1]
+      df[seq(from=index_of_NA[4], to=(index_of_NA[5]-1)) , 9] <- df[index_of_NA[4],1]
+      df[seq(from=index_of_NA[5], to=(index_of_NA[6]-1)) , 9] <- df[index_of_NA[5],1]
+      df[seq(from=index_of_NA[6], to=(index_of_NA[7]-1)) , 9] <- df[index_of_NA[6],1]
+      df[seq(from=index_of_NA[7], to=(index_of_NA[8]-1)) , 9] <- df[index_of_NA[7],1]
+      df[seq(from=index_of_NA[8], to=(index_of_NA[9]-1)) , 9] <- df[index_of_NA[8],1]
+      df[seq(from=index_of_NA[9], to=(nrow(df))) , 9] <- df[index_of_NA[9],1]
+
+      df$unit <- NA
+      df <- as.data.frame(df)
+
+      df[seq(from=index_of_NA[1], to=(index_of_NA[2]-1)) , 10] <- df[index_of_NA[2],2]
+      df[seq(from=index_of_NA[2], to=(index_of_NA[3]-1)) , 10] <- df[index_of_NA[3],2]
+      df[seq(from=index_of_NA[3], to=(index_of_NA[4]-1)) , 10] <- df[index_of_NA[3],2]
+      df[seq(from=index_of_NA[4], to=(index_of_NA[5]-1)) , 10] <- df[index_of_NA[4],2]
+      df[seq(from=index_of_NA[5], to=(index_of_NA[6]-1)) , 10] <- df[index_of_NA[5],2]
+      df[seq(from=index_of_NA[6], to=(index_of_NA[7]-1)) , 10] <- df[index_of_NA[6],2]
+      df[seq(from=index_of_NA[7], to=(index_of_NA[8]-1)) , 10] <- df[index_of_NA[7],2]
+      df[seq(from=index_of_NA[8], to=(index_of_NA[9]-1)) , 10] <- df[index_of_NA[8],2]
+      df[seq(from=index_of_NA[9], to=(nrow(df))) , 10] <- df[index_of_NA[9],2]
+
+      df <- df[!is.na(df$'2030'), ]
+
+      df[,2] <- as.numeric(df[,2])
+      df[,3] <- as.numeric(df[,3])
+      df[,5] <- as.numeric(df[,5])
+      df[,6] <- as.numeric(df[,6])
+      df[,8] <- as.numeric(df[,8])
+
+      dfp <- pivot_longer(df, cols = c(2:8))
+
+      dfp$period <- NA
+      dfp <- as.data.frame(dfp)
+      dfp[seq(from=1, to=nrow(dfp), by =7) , 6] <- "2020"
+      dfp[seq(from=2, to=nrow(dfp), by =7) , 6] <- "2030"
+      dfp[seq(from=3, to=nrow(dfp), by =7) , 6] <- "2030"
+      dfp[seq(from=4, to=nrow(dfp), by =7) , 6] <- "2030"
+      dfp[seq(from=5, to=nrow(dfp), by =7) , 6] <- "2050"
+      dfp[seq(from=6, to=nrow(dfp), by =7) , 6] <- "2050"
+      dfp[seq(from=7, to=nrow(dfp), by =7) , 6] <- "2050"

```

Comment:
it would be better not to have any hard-coded data. suggested change: read the values from the data

Reply:
the excel has Current for 2020 and Ultimate for 2050 and I use the function sub